daml-script/daml/Daml/Script.daml

```diff
@@ -579,6 +587,9 @@ data LedgerValue = LedgerValue {}
 fromLedgerValue : LedgerValue -> a
 fromLedgerValue = error "foobar" -- gets replaced by the identity-function in script/Runner.scala
 
+toLedgerValue : a -> LedgerValue
```

Comment:
I needed this to write the `CatchPayload` in `instance ActionCatch Script`. Since it's inside `lift $ Free $ Catch $ _`, it needs to have type `CatchPayload (Free ScriptF t)` (where `t` is the return type of the `Script`). For context this is what the datatype looks like (in this PR):

```haskell
data CatchPayload a = CatchPayload
  with
    act : () -> Free ScriptF (Free ScriptF LedgerValue, ())
    handle : AnyException -> Optional (Free ScriptF LedgerValue)
    continue : Free ScriptF LedgerValue -> a
```

really, we would like to say that the `LedgerValue` returned by `act`, the `LedgerValue` possibly returned by `handle`, and the `LedgerValue` consumed by `continue` are all the same type, let's call it `x`, for something like

```haskell
data CatchPayload x a = CatchPayload
  with
    act : () -> Free ScriptF (Free ScriptF x, ())
    handle : AnyException -> Optional (Free ScriptF x)
    continue : Free ScriptF x -> a
```

the problem with that is that then it's not clear what to put in for `x` in the `Catch` constructor of `ScriptF`:

```haskell
data ScriptF a 
  = Submit (SubmitCmd a)
  | {- ... -}
  | Catch (CatchPayload (???) a)
```

what we would want here is to say any type is fine, since what really matters is that all three fields agree on what it is. In Haskell we could use an existential type for exactly that, but those aren't allowed in Daml. However we do have `LedgerValue`, a fake type representing a value of _some_ type, which is pretty close to what we want. Since we're the ones constructing the `CatchPayload`, we can make sure that they agree, and then `toLedgerValue` lets us convince the typechecker.

Reply:
prompted by this I've changed the encoding a bit. Now `CatchPayload` is defined as above with the `x` argument, and instead of constructing it by using `{from,to}LedgerValue` in each field, we construct it with type `CatchPayload t (Free ScriptF t)` and then cast it to `CatchPayload LedgerValue (Free ScriptF t)` using a new fake function `castCatchPayload : CatchPayload x a -> CatchPayload LedgerValue a` (using the same `id` trick). This removes the need for `toLedgerValue` and gives us a bit more confidence that the types make sense before the cast.